influxdb3_write/src/write_buffer/mod.rs

```diff
@@ -1135,6 +1135,56 @@ pub(crate) enum PluginEvent {
 
 impl WriteBuffer for WriteBufferImpl {}
 
+pub async fn check_mem_and_force_snapshot_loop(
+    write_buffer: Arc<WriteBufferImpl>,
+    memory_threshold_bytes: usize,
+    check_interval: Duration,
+) -> tokio::task::JoinHandle<()> {
+    tokio::spawn(async move {
+        let mut interval = tokio::time::interval(check_interval);
+        interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+        loop {
+            interval.tick().await;
+            check_mem_and_force_snapshot(&write_buffer, memory_threshold_bytes).await;
+        }
+    })
+}
+
+async fn check_mem_and_force_snapshot(
+    write_buffer: &Arc<WriteBufferImpl>,
+    memory_threshold_bytes: usize,
+) {
+    let current_buffer_size_bytes = write_buffer.buffer.get_total_size_bytes();
+    debug!(
+        current_buffer_size_bytes,
+        memory_threshold_bytes, "checking buffer size and snapshotting"
+    );
+
+    if current_buffer_size_bytes >= memory_threshold_bytes {
+        info!(
```

Comment:
I'd have this as a `warn`. Ideally, we won't be forcing snapshots. If we are, it means there's memory pressure or configuration settings that should probably be changed.